server/svix-server/src/v1/endpoints/auth.rs

```diff
@@ -62,21 +69,36 @@ impl From<DashboardAccessOut> for AppPortalAccessOut {
 /// Use this function to get magic links (and authentication codes) for connecting your users to the Consumer Application Portal.
 #[aide_annotate(op_id = "v1.authentication.app-portal-access")]
 async fn app_portal_access(
-    State(AppState { cfg, .. }): State<AppState>,
+    State(AppState { cfg, ref cache, .. }): State<AppState>,
     _: Path<ApplicationPath>,
     permissions::OrganizationWithApplication { app }: permissions::OrganizationWithApplication,
     ValidatedJson(data): ValidatedJson<AppPortalAccessIn>,
 ) -> Result<Json<AppPortalAccessOut>> {
+    let one_time_token = generate_one_time_token()?;
+    let key = OneTimeTokenCacheKey::new(&one_time_token);
+
+    // The "exchange" endpoint will give this JWT back when requested with the one time token.
     let token = generate_app_token(
         &cfg.jwt_secret,
         app.org_id,
         app.id.clone(),
         data.feature_flags,
     )?;
-
+    ctx!(
+        cache
+            .set_string(
+                &key,
+                &token,
+                // FIXME: what duration?
+                Duration::from_secs(60 * 60 * 24 * 7)
+            )
+            .await
+    )?;
     let login_key = serde_json::to_vec(&serde_json::json!({
         "appId": app.id,
-        "token": token,
+        // FIXME: is it okay to change this key? Seems like it's a breaking change.
+        //   Do we need to support both one time and plain jwt?
```

Comment:
This is what we output, and I think it's fine as we don't promise the URL structure to be part of the API. Especially not the base64 encoded part!